### DISCUSSION TITLE
What is the difference between

### DISCUSSION THREAD

**Discussion by mr-qansuh** [staxrip/staxrip#878] [Q&A] [answered]
What is the difference between  Directshowsource and FFvideosource and LWLibavideosource
and if i use directshowsource 
Will there be a affect in quality?

**Comment by jessielw** [comment, marked as answer]
DirectShowSource (http://avisynth.nl/index.php/DirectShowSource)
FFmpegSource (http://avisynth.nl/index.php/FFmpegSource)
LSMASHSource (http://avisynth.nl/index.php/LSMASHSource) [LWLibavideosource]

Most people use FFmpegSource, as it indexes for frame accuracy.

**Comment by mr-qansuh** [comment]
> l there be a affect in quality?

if i use directshowsource
Will there be a affect in quality?

**Comment by jessielw** [comment]
There is no difference in quality. However dirsctshowsource has limited container support. 
See here https://stackoverflow.com/questions/48323121/difference-between-directshowsource-and-ffmpegsource2-in-avisynth

Ffms2 indexes for frame accurate seeking. This is for avisynth and has nothing to do with the quality. It writes a physical index file next to the video for this. 

See here for more information 
https://forum.videohelp.com/threads/383449-FFVideoSource-vs-LWLibavVideoSource

Direct show source might not even support your container. Use libwav or ffvideosource.
